src/tls/boring.rs

```diff
@@ -309,6 +309,7 @@ pub fn grpc_connector(
     http.set_keepalive(Some(Duration::from_secs(300)));
     http.set_keepalive_interval(Some(Duration::from_secs(75)));
     http.set_keepalive_retries(Some(9));
+    http.set_connect_timeout(Some(Duration::from_millis(100)));
```

Comment:
we probably want 10s, not 100ms. 100ms is actually pretty quick for a connection, especially cross-zone

Reply:
per @hzxuzhonghu's comment, it looks like we may use the default from the envoy `go-control-plane` which is `5s`. I'm not supremely confident in that but don't see where we set a connect timeout. Curious to be proven wrong if I missed it though. How do we feel about `5s`?

Reply:
Its not from that library but still 5s: https://github.com/istio/istio/blob/4482b870e91ce66b6cc776700cb4c94f1edc8dca/pkg/istio-agent/xds_proxy.go#L334. 5s is good.